File: problems/1658/solution.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        int ans = INT_MAX, cur = 0, left = 0, n = nums.size();
        
        for (int num: nums) {
            cur += num;
        }
        
        for (int right = 0; right < nums.size(); right++) {
            cur -= nums[right];
            
            while (left <= right && cur < x) {
                cur += nums[left++];
            }
            
            if (cur == x) {
                ans = min(ans, n - right + left - 1);
            }
        }
        
        return ans == INT_MAX ? -1 : ans;
    }
};

File: problems/1658/solution.go
package main
const MaxInt = int(^uint(0) >> 1)

func minOperations(nums []int, x int) int {
    ans, sum, n := MaxInt, 0, len(nums)
    
    for _, num := range nums {
        sum += num
    }
    
    for left, right := 0, 0; right < n; right++ {
        sum -= nums[right]
        
        for left <= right && sum < x {
            sum += nums[left]
            left++
        }
        
        if (sum == x) {
            ans = min(ans, n - (right - left) - 1)
        }
    }
    
    if (ans == MaxInt) {
        return -1
    }
    
    return ans
}

func min(a, b int) int {
    if a < b {
        return a
    }
    
    return b
}